tools/witx/src/render.rs

```diff
@@ -0,0 +1,266 @@
+                SExpr::annot("witx"),
+                SExpr::word("const_pointer"),
+                p.to_sexpr(),
+            ]),
+            DatatypeIdent::Ident(i) => i.name.to_sexpr(),
+        }
+    }
+}
+
+impl Render for Datatype {
+    fn to_sexpr(&self) -> SExpr {
+        let name = self.name.to_sexpr();
+        let body = self.variant.to_sexpr();
+        SExpr::Vec(vec![SExpr::word("typename"), name, body])
+    }
+}
+
+impl Render for DatatypeVariant {
+    fn to_sexpr(&self) -> SExpr {
+        match self {
+            DatatypeVariant::Alias(a) => a.to_sexpr(),
+            DatatypeVariant::Enum(a) => a.to_sexpr(),
+            DatatypeVariant::Flags(a) => a.to_sexpr(),
+            DatatypeVariant::Struct(a) => a.to_sexpr(),
+            DatatypeVariant::Union(a) => a.to_sexpr(),
+        }
+    }
+}
+
+impl Render for AliasDatatype {
+    fn to_sexpr(&self) -> SExpr {
+        self.to.to_sexpr()
+    }
+}
+
+impl Render for EnumDatatype {
+    fn to_sexpr(&self) -> SExpr {
+        let header = vec![SExpr::word("enum"), self.repr.to_sexpr()];
+        let variants = self
+            .variants
+            .iter()
+            .map(|v| v.to_sexpr())
+            .collect::<Vec<SExpr>>();
+        SExpr::Vec([header, variants].concat())
+    }
+}
+
+impl Render for FlagsDatatype {
+    fn to_sexpr(&self) -> SExpr {
+        let header = vec![SExpr::word("flags"), self.repr.to_sexpr()];
+        let flags = self
+            .flags
+            .iter()
+            .map(|f| SExpr::Vec(vec![SExpr::word("flag"), f.to_sexpr()]))
+            .collect::<Vec<SExpr>>();
+        SExpr::Vec([header, flags].concat())
+    }
+}
+
+impl Render for StructDatatype {
+    fn to_sexpr(&self) -> SExpr {
+        let header = vec![SExpr::word("struct")];
+        let members = self
+            .members
+            .iter()
+            .map(|m| {
+                SExpr::Vec(vec![
+                    SExpr::word("field"),
+                    m.name.to_sexpr(),
+                    m.type_.to_sexpr(),
+                ])
+            })
+            .collect::<Vec<SExpr>>();
+        SExpr::Vec([header, members].concat())
+    }
+}
+
+impl Render for UnionDatatype {
+    fn to_sexpr(&self) -> SExpr {
+        let header = vec![SExpr::word("union")];
+        let variants = self
+            .variants
+            .iter()
+            .map(|v| {
+                SExpr::Vec(vec![
+                    SExpr::word("field"),
+                    v.name.to_sexpr(),
+                    v.type_.to_sexpr(),
+                ])
+            })
+            .collect::<Vec<SExpr>>();
+        SExpr::Vec([header, variants].concat())
+    }
+}
+
+impl Render for IntRepr {
+    fn to_sexpr(&self) -> SExpr {
+        match self {
+            IntRepr::I8 => SExpr::word("u8"),
+            IntRepr::I16 => SExpr::word("u16"),
+            IntRepr::I32 => SExpr::word("u32"),
+            IntRepr::I64 => SExpr::word("u64"),
```

Comment:
Since we're now round-tripping `IntRepr` from "u" type to "u" type, it seems like we should rename `IntRepr::I8` etc. to `IntRepr::U8` etc.

Reply:
Agree, will change this